torch/_inductor/utils.py

```diff
@@ -44,6 +44,20 @@
 import sympy
 
 import torch
+
+
+GPU_TYPES = ["cuda", "xpu"]
+
+
+def _get_gpu_type():
+    avail_gpus = [x for x in GPU_TYPES if getattr(torch, x).is_available()]
+    assert len(avail_gpus) <= 1
+    gpu_type = "cuda" if len(avail_gpus) == 0 else avail_gpus.pop()
+    return gpu_type
+
+
+GPU_TYPE = _get_gpu_type()
```

Comment:
Please don't create global `GPU_TYPE` environment variable, but rather memoize results of `_get_gpu_type()`, otherwise importing inductor would cause GPU subsystem initialization even if user only plans to use inductor for CPU 
```suggestion
GPU_TYPES = ["cuda", "xpu"]


@functools.lru_cache(None)
def _get_gpu_type():
    avail_gpus = [x for x in GPU_TYPES if getattr(torch, x).is_available()]
    assert len(avail_gpus) <= 1
    gpu_type = "cuda" if len(avail_gpus) == 0 else avail_gpus.pop()
    return gpu_type


```

Reply:
> Please don't create global `GPU_TYPE` environment variable, but rather memoize results of `_get_gpu_type()`, otherwise importing inductor would cause GPU subsystem initialization even if user only plans to use inductor for CPU
Thanks for the reminder.